### ISSUE TITLE
HPA not working with auto-provisioned metric server on AKS 1.14.6

### ISSUE THREAD

**Issue by stephaneey** [Azure/AKS#1190]
**What happened**:
HPA not working with brand new cluster
**What you expected to happen**:
HPA should work out of the box 
**How to reproduce it (as minimally and precisely as possible)**:
When creating a brand new cluster 
az aks create --name aks5 --resource-group aks5 --node-count 1 --generate-ssh-keys --aad-server-app-id <anapp>--aad-server-app-secret <asecret> --aad-client-app-id <anapp> --aad-tenant-id <tenant> --kubernetes-version 1.14.6
metrics-server is self-provisioned, so far so good. 
Deploying a custom POD with the following HPA:
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  name: mserverhpatest
  namespace: default
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: mserverhpatest
  minReplicas: 1
  maxReplicas: 10
  metrics:
  - type: Resource
    resource:
      name: cpu
      target:
        type: Utilization
        averageUtilization: 5
always end up with no metrics found. 

Indeed:
kubectl get hpa                                                                                                                                                                NAME             REFERENCE                   TARGETS        MINPODS   MAXPODS   REPLICAS   AGE
mserverhpatest   Deployment/mserverhpatest   **<unknown>**/5%   1         10        1          2m56s

checking metrics-server logs shows:

No metrics for pod default/mserverhpatest-8485f4c4d8-ncrrp

although kubectl top pod returns this:

NAME                              CPU(cores)   MEMORY(bytes)
mserverhpatest-8485f4c4d8-ncrrp   0m           37Mi

Going endlessly from Github issue to github issue, It seems that this is due to an invalid configuration of metrics-server (provisioned by AKS itself since I didn't install it myself). Any attempt to modify the deployment results in metrics-server self resuming to its previous configuration.

**Anything else we need to know?**:

**Environment**:
version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
- Size of cluster (how many worker nodes are in the cluster?)
1 node

**Comment by cpunella** [comment]
Hi,

quick tip: have you set resource limit in the deployment yaml definition? There is a correlation between metrics and pod limit definitions...

**Comment by stephaneey** [comment]
Hello,

Thanks for your answer and yes...it's my bad! I had deployed the resources like this:
resources: 
         limits:
          cpu: "1"
         requests:
          cpu: "0.5"
but when preparing a copy/paste for the issue, I noticed that further on in the same payload, there was another resource element like this: resources: {} which invalidated the former resource declaration. Didn't notice this one! Sorry & thanks for your answer :)
